crates/core/src/models/event.rs

```diff
@@ -87,3 +85,75 @@ impl ConvertibleStarknetEvent for StarknetEvent {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::client::tests::init_client;
+    use crate::mock::mock_starknet::fixtures;
+
+    #[test]
+    fn test_to_eth_log_log3() {
+        // Given
+        let event: Event = serde_json::from_str(include_str!("test_data/conversion/starknet/event_log3.json")).unwrap();
+        let starknet_event = StarknetEvent::new(event);
+
+        let fixtures = fixtures(vec![]);
+        let client = init_client(Some(fixtures));
+
+        // When
+        let eth_log = starknet_event.to_eth_log(&client, None, None, None, None, None).unwrap();
+
+        // Then
+        let expected: Log = serde_json::from_str(include_str!("test_data/conversion/eth/event_log3.json")).unwrap();
+        assert_eq!(expected, eth_log);
+    }
+
+    #[test]
+    fn test_to_eth_log_log4() {
+        // Given
+        let event: Event = serde_json::from_str(include_str!("test_data/conversion/starknet/event_log4.json")).unwrap();
+        let starknet_event = StarknetEvent::new(event);
+
+        let fixtures = fixtures(vec![]);
+        let client = init_client(Some(fixtures));
+
+        // When
+        let eth_log = starknet_event.to_eth_log(&client, None, None, None, None, None).unwrap();
+
+        // Then
+        let expected: Log = serde_json::from_str(include_str!("test_data/conversion/eth/event_log4.json")).unwrap();
+        assert_eq!(expected, eth_log);
+    }
+
+    #[test]
+    #[should_panic(expected = "KakarotDataFilteringError(\"Event\")")]
```

Comment:
this cool AF!

Reply:
yes love it, it was pointed out by @ClementWalter in a previous PR that we should be using this!